File: strings/3541-find-most-frequent-vowel-and-consonant.go
package strings

// https://leetcode.com/problems/find-most-frequent-vowel-and-consonant/

func maxFreqSum(s string) int {
	var freq [26]int
	for _, c := range s {
		freq[int(c-'a')]++
	}

	vowels := map[rune]bool{'a': true, 'e': true, 'i': true, 'o': true, 'u': true}

	var vmx, cmx int

	for i, f := range freq {
		if vowels[rune('a'+i)] {
			vmx = max(vmx, f)
		} else {
			cmx = max(cmx, f)
		}
	}

	return vmx + cmx
}

File: strings/3541-find-most-frequent-vowel-and-consonant.py
from collections import Counter
import unittest

# https://leetcode.com/problems/find-most-frequent-vowel-and-consonant/

# python3 -m unittest strings/3541-find-most-frequent-vowel-and-consonant.py


class Solution(unittest.TestCase):
    def maxFreqSum(self, s: str) -> int:
        freq = Counter(s)
        vowels = set(['a', 'e', 'i', 'o', 'u'])
        vmx = max((f for c, f in freq.items() if c in vowels), default=0)
        cmx = max((f for c, f in freq.items() if c not in vowels), default=0)
        return vmx + cmx

    def test(self):
        for s, expected in [
            ("successes", 6),
            ("aeiaeia", 3),
        ]:
            output = self.maxFreqSum(s)
            self.assertEqual(expected, output, f"expected: {expected}, output: {output}")
